src/active_query.rs

```diff
@@ -45,7 +45,7 @@ pub(crate) struct ActiveQuery {
     /// This table starts empty as the query begins and is gradually populated.
     /// Note that if a query executes in 2 different revisions but creates the same
     /// set of tracked structs, they will get the same disambiguator values.
-    disambiguator_map: FxHashMap<u64, Disambiguator>,
+    disambiguator_map: FxHashMap<(IngredientIndex, u64), Disambiguator>,
```

Comment:
Does this change allow us to remove the DatabaseKeyIndex from the id map below because the index can now be queried from the key?

Reply:
No, its different index, afaik. IngredientIndex is kinda TypeId for tracked struct